frontend/src/components/intake/ShasIntakeSchema.ts

```diff
@@ -0,0 +1,144 @@
+        otherwise: () => string().nullable()
+      }),
+      otherSelected: boolean(),
+      otherUnitsDescription: string().when('otherSelected', {
+        is: (prev: boolean) => prev,
+        then: (schema) => schema.required().label('Description of units'),
+        otherwise: () => string().nullable()
+      }),
+      otherUnits: string().when('otherSelected', {
+        is: (prev: boolean) => prev,
+        then: (schema) => schema.required().oneOf(NumResidentialUnits).label('Expected number of other units'),
+        otherwise: () => string().nullable()
+      })
+    }).test('housing-checkbox-test', 'At least one residential type must be selected', (value, context) => {
+      return (
+        context.originalValue.singleFamilySelected ||
+        context.originalValue.multiFamilySelected ||
+        context.originalValue.otherSelected
+      );
+    }),
+    location: object({
+      projectLocation: string().required().label('Location'),
+      streetAddress: string().when('projectLocation', {
+        is: (prevQuestion: string) => prevQuestion === ProjectLocation[0],
+        then: () => stringRequired.label('Street address'),
+        otherwise: () => string().nullable()
+      }),
+      locality: string().when('projectLocation', {
+        is: (prevQuestion: string) => prevQuestion === ProjectLocation[0],
+        then: () => stringRequired.label('Locality'),
+        otherwise: () => string().nullable()
+      }),
+      province: string().when('projectLocation', {
+        is: (prevQuestion: string) => prevQuestion === ProjectLocation[0],
+        then: () => stringRequired.label('Province'),
+        otherwise: () => string().nullable()
+      }),
+      latitude: number().when('projectLocation', {
+        is: (prevQuestion: string) => prevQuestion === ProjectLocation[1],
+        then: () => number().required().min(48).max(60).label('Latitude'),
+        otherwise: () => string().nullable()
+      }),
+      longitude: number().when('projectLocation', {
+        is: (prevQuestion: string) => prevQuestion === ProjectLocation[1],
+        then: () => number().required().min(-139).max(-114).label('Longitude'),
+        otherwise: () => string().nullable()
+      }),
+      ltsaPIDLookup: string().max(255).label('Parcel ID'),
+      geomarkUrl: string().max(255).label('Geomark web service url')
+    }),
+    permits: object({
+      hasAppliedProvincialPermits: string().oneOf(YesNoUnsure).required().label('Applied permits'),
+      checkProvincialPermits: string().when('hasAppliedProvincialPermits', {
+        is: (prevQuestion: string) => prevQuestion === BASIC_RESPONSES.YES || prevQuestion === BASIC_RESPONSES.UNSURE,
+        then: (schema) => schema.oneOf(YesNo).required().label('Check permits'),
+        otherwise: (schema) => schema.nullable()
+      })
+    }),
+    appliedPermits: array().of(
+      object({
+        permitTypeId: number().required().max(255).label('Permit type'),
```

Comment:
Should not have `max` here. While it's entirely unlikely the permit type IDs will go beyond this it could happen.